### PR TITLE
DEPR-129: Remove EdxRestApiClient usage in ecommerce-scripts

### PR DESCRIPTION
**Pull request by UvgenGen**
JIRA: https://openedx.atlassian.net/browse/DEPR-129
YouTrack: https://youtrack.raccoongang.com/issue/OeX_Depr-28

### CODE REVIEW

File: course_discovery/populate_course_run_reporting_type_and_announcement.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-import datetime
-import json
 import logging
 import os
 
 from dateutil.parser import parse
 
 from services import CatalogApiService
 
+USER_AGENT = os.environ.get('API_CLIENT_NAME', 'ecommerce-scripts')
```
Comment (idegtiarov) [inline course_discovery/populate_course_run_reporting_type_and_announcement.py]: Do we expect any other user agent but ecommerce-scripts that could be used in our code?

File: course_discovery/services.py [hunk 2]
```diff
@@ -1,25 +1,44 @@
...
+                    'token_type': 'jwt',
+                },
+                headers={
+                    'User-Agent': user_agent,
+                },
+                timeout=(3.1, 5)
             )
+            data = response.json()
+            self.access_token = data['access_token']
         except Exception:
```
Comment (idegtiarov) [inline course_discovery/services.py:35]: In the new code, we have three potential points to raise an Exception lets have every covered with the except case.

File: course_discovery/services.py [hunk 3]
```diff
@@ -1,25 +1,44 @@
...
 
-        self.api_client = EdxRestApiClient(api_url_root, jwt=self.access_token)
+    def update_course_run(self, key, data):
+        headers = {'Authorization': "JWT {}".format(self.access_token)}
+        response = requests.patch(
+            f'{self.api_url_root}course_runs/{key}/',
+            json=data,
+            headers=headers
+        )
+        data = response.json()
```
Comment (idegtiarov) [inline course_discovery/services.py]: Why do we need to handle the data in this method? Why do we need this method at all?

File: upgrade_deadlines/repair.py [hunk 4]
```diff
@@ -6,39 +6,59 @@
...
+            'client_secret': OAUTH_SECRET,
+            'token_type': 'jwt',
+        },
+        headers={
+            'User-Agent': USER_AGENT,
+        },
+        timeout=(3.1, 5)
+    )
+    data = response.json()
+    return data['access_token']
```
Comment (idegtiarov) [inline upgrade_deadlines/repair.py]: It seems that potentials exceptions for making requests as well as for taking a value for the key are not handled, should it be added somewhere?

File: upgrade_deadlines/repair.py [hunk 5]
```diff
@@ -6,39 +6,59 @@
...
+
+
+def get_course_runs(jwt, querystring):
+    headers = {'Authorization': "JWT {}".format(jwt)}
+    response = requests.get(
```
Comment (idegtiarov) [inline upgrade_deadlines/repair.py:54]: ditto

**Review by idegtiarov — CHANGES_REQUESTED** [review]
Please take a look at my comments online.

File: course_discovery/services.py (hunk 2 above)
Comment (UvgenGen) [reply course_discovery/services.py:35]: updated

File: upgrade_deadlines/repair.py (hunk 5 above)
Comment (UvgenGen) [reply upgrade_deadlines/repair.py:54]: updated

File: upgrade_deadlines/repair.py (hunk 4 above)
Comment (UvgenGen) [reply upgrade_deadlines/repair.py]: added handling exceptions

File: course_discovery/services.py (hunk 3 above)
Comment (UvgenGen) [reply course_discovery/services.py]: removed

File: course_discovery/populate_course_run_reporting_type_and_announcement.py (hunk 1 above)
Comment (UvgenGen) [reply course_discovery/populate_course_run_reporting_type_and_announcement.py]: No, we don't expect. Removed.

File: upgrade_deadlines/repair.py [hunk 6]
```diff
@@ -25,20 +25,49 @@
...
+        access_token = data['access_token']
+    except (KeyError, json.decoder.JSONDecodeError) as json_error:
+        raise requests.RequestException(response=response) from json_error
+    return access_token
+
+
+def get_course_runs(jwt, querystring):
+    headers = {'Authorization': "JWT {}".format(jwt)}
+    response = requests.get(
+        f'{DISCOVERY_API_URL}course_runs/',
```
Comment (dyudyunov) [inline upgrade_deadlines/repair.py:55]: Possible `/` issue (discovery_api_url might be without trailing slash)

JFYI

File: course_discovery/services.py (hunk 2 above)
Comment (idegtiarov) [reply course_discovery/services.py:35]: My thought was not to shadow exceptions inside the try - except block but to add additional checks in the external block as well as a different error logging, please take a  look at the example:
```python
try:
    ...
    data = response.json()
    self.access_token = data['access_token']
except (KeyError, json.decoder.JSONDecodeError) as json_error:
     logger.exception()
     raise
except Exception:
     ...
```

**Review by idegtiarov — CHANGES_REQUESTED** [review]
One minor nit. Other LGTM

**Review by idegtiarov — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**